### ISSUE TITLE
Find Controllers option showing only one controller connected

### ISSUE THREAD

**Issue by jang2ah2** [ndeadly/MissionControl#155]
Hi,

Just wanna report a minor issue. 
I updated to Atmosphere v0.16.0 and used MissionControl v0.4 then I used the "Disconnect Controllers" to refresh all paired controllers.
everything works Great.
got 2 Ps4 Controllers paired already. 

but every time I put my switch to sleep and wake it up, there's always a notification saying that joycon paired or controller paired. 
and then only one controller showing on Find Controllers (either left or right joycon, or a controller) 

I play around this option connecting, disconnecting, sleep, and reset. and then not sure what the exact step i did but got an error message saying that I reach the maximum limit for connecting controllers. 

not a game breaker for me. but just wanna know if this just me or maybe happened to everyone else.

Thanks!!!

edit: [additional]
this happened under Emummc firmware v10.2.0 with Atmosphere v0.16.0 same issue with sysnand v11 with Atmosphere v0.16.0

and when i tried to remove mission control. 2 joycons showing under Find Controller just like it supposed to be.

**Comment by jang2ah2** [comment]
I manage to get the error message i was saying. 
what i did is, i just go the the Find Controller Screen, remove the 2 joycons and then put it back both joycons again. 
![2020120620261000-1E95E5926F1CB99A87326D927F27B47E](https://user-images.githubusercontent.com/71596703/101280136-29b2fb00-377c-11eb-8e8a-ae3979e9c932.jpg)

**Comment by ndeadly** [comment]
Hmm that's a strange one I haven't heard of before. Usually you get those notifications directly after clearing the database if your joycons are attached to the console.

My guess is that this has something to do with pairing via the joycon rails, possibly additional confusion due to the use of emuMMC. Your sysNAND and emuMMC both have their own copy of the pairing database, and those are only synchronised when you first create your emuMMC. After that if you do anything involving controller pairing, they no longer contain the same information for all controllers. A controller may be present in both databases, but only have accurate pairing information in one of them. In addition, when pairing happens via the rails (or USB cable in the case of the Pro Controller) the data stored differs from what gets stored when you pair via Bluetooth. I have seen this cause some strange things to happen in the past.

I would try going through the database deleting and re-pairing process again via Bluetooth. Make sure your controllers aren't physically connected the the console or else they will instantly re-pair via the rails. You may even want to use the native option in the settings menu to delete the database rather than btdb.nro in case there's some subtle difference. When paired properly via Bluetooth, controllers should show a vendor ID of `057e`

**Comment by jang2ah2** [comment]
Now i tried to switch back to my previous Atmosphere v14.4 with MissionControl v0.3
And when I check the Find Controllers Option. it's showing all controllers I connected.
![2020120621062800-1E95E5926F1CB99A87326D927F27B47E](https://user-images.githubusercontent.com/71596703/101281766-da71c800-3785-11eb-8e72-11a00a817247.jpg)


And When I updated back to Atmosphere v16 with MissionControl v0.4
back to same issue. only one controller showing on Find Controllers screen. although all controllers are paired. 
and i get the same error message from my previous post. 
![2020120621315000-1E95E5926F1CB99A87326D927F27B47E](https://user-images.githubusercontent.com/71596703/101281781-03925880-3786-11eb-8ac4-2c2f7fe91fcc.jpg)


maybe this is just an isolated case. all controllers works anyway. so still good for me.

**Comment by ndeadly** [comment]
Did doing what I suggested above make any difference?

Which setup are you trying this on? sysNAND or emuMMC? Which HOS version? I'll try and reproduce it later if I can. I can't think of anything that changed between 0.3.0 and 0.4.0 that would affect this, but it definitely shouldn't be happening.

**Comment by jang2ah2** [comment]
yep. I did clear the paired controllers multiple times using SystemSettings>Controllers&Sensors>DisconnectControllers.

Regardless if i used sysnand or emummc. as soon as i used the MCv0.4 this happened. and I did uninstall MCv0.4 and try to check the FindControllers. both joycon is showing right away. and when I installed back MCv0.4 same thing happened. only one controller showing.

but all controllers is working fine.

**Comment by ndeadly** [comment]
Hmm ok, wasn't clear from your initial post whether you were a new user or had used a previous version. There must be a subtle bug introduced somewhere between versions then if that's the case. I did have to rewrite a bunch of stuff for this release in order to switch over to the latest libnx stuff that clashed with my own bluetooth code. I will look into it. Let me know if you find anything else affected by it.

**Comment by jang2ah2** [comment]
Sure thing. Glad that I could help in someways. As soon as i get other unusual behavior, I'll let you know. 
Cheers.

**Comment by Roxas-Dev** [comment]
This is happening to me as well, except the left joystick isn't working in games but outside of them for some odd reason it does and on rebooting to payload it forgets the button remapping. ( Dualshock 4 ) also tried repairing various times but issue persists and was working fine on the previous version.

**Comment by ndeadly** [comment]
@Roxas-Dev  the dualshock4 analog stick problem was due to a copy/paste bug in the initial release of 0.4.0 that I fixed shortly after release without bumping the version number. Re-downloading the 0.4.0 release should fix it.

**Comment by Roxas-Dev** [comment]
@ndeadly awesome thanks for letting me know.

**Comment by ndeadly** [comment]
@jang2ah2 I've managed to reproduce everything you described on my own 11.0.0 console by deleting the pairing database. Now to try and figure out why it's happening.

**Comment by ndeadly** [comment]
Alright, think I've found it. Seems to have been the result of confusion caused by libnx naming a couple of structs differently to what I had used prior to updating. The wrong data type ended up being passed to a function I'm intercepting, so the output was being truncated. I'm surprised this didn't crash anything.

See if this works for you guys
[MissionControl-0.4.0-btm-bugfix.zip](https://github.com/ndeadly/MissionControl/files/5655473/MissionControl-0.4.0-btm-bugfix.zip)
